rtl/eclrtl/rtlkey.cpp

```diff
@@ -24,186 +24,101 @@
 #include "rtlrecord.hpp"
 #include "rtlnewkey.hpp"
 
-#define KSM_SET             0x01
-#define KSM_WILD            0x02
-#define KSM_LITTLE_ENDIAN   0x04
-#define KSM_SIGNED          0x08
-#define KSM_VAROFFSET       0x10
-#define KSM_TRANSLATED      0x20
-
-
-
 class CKeySegmentMonitor : implements IKeySegmentMonitor, public CInterface
 {
 protected:
     size32_t size;
     size32_t offset;
+    unsigned fieldIdx;
     unsigned hash;
 
 public:
     IMPLEMENT_IINTERFACE;
-    CKeySegmentMonitor(unsigned _offset, unsigned _size);
+    CKeySegmentMonitor(unsigned _fieldIdx, unsigned _offset, unsigned _size);
     CKeySegmentMonitor(MemoryBuffer &mb)
     {
-        mb.read(size).read(offset).read(hash);
+        mb.read(size).read(offset).read(fieldIdx).read(hash);
```

Comment:
Does this mean that the remote index code added in 6.4 cannot be used from 7.0?  (I'm not sure if that was the plan anyway).